File: js/dart/g20.super/Student.dart
import "Person.dart";

/******* CLASS DECLARATION *******/
class Student extends Person {
  double _gpa;

  /******* CONSTRUCTORS *******/
  Student(String name, double gpa) : super(name) {
    this.gpa = gpa;
  }
  
  Student.fromName (String name) : super(name) { /* nothing */ }

  /******* PROPERTIES *******/
  get gpa => _gpa;
  set gpa(value) => _gpa = value;
}

File: js/js/g20.super/Student.js
load('Person.js');
/******* CLASS DEFINITION *******/
var Student = (function () {
  /******* CONSTRUCTOR *******/
  function Student (name, gpa) {
    Person.prototype.Person.call(this) = name;
    this.gpa = gpa != null ? gpa : 0.0;
  }

  /******* PROPERTIES *******/
  Object.defineProperty(Student.prototype, "gpa", {
    get: function(){ return this._gpa },
    set: function(value) { this._gpa = value }
  });

  return Student;
})(Person);
